src/test/java/com/faforever/client/game/GameServiceTest.java

```diff
@@ -669,6 +670,24 @@ public void startSearchMatchmaker() throws IOException {
         gameLaunchMessage.getLeaderboard(), GPG_PORT, LOCAL_REPLAY_PORT, false, junitPlayer);
   }
 
+  @Test
+  public void startSearchMatchmakerWithGameOptions() throws IOException {
+    when(preferencesService.isGamePathValid()).thenReturn(true);
+    when(modService.getFeaturedMod(FAF.getTechnicalName()))
+        .thenReturn(completedFuture(FeaturedModBeanBuilder.create().defaultValues().get()));
+    Map<String, String> gameOptions = new HashMap<>();
+    gameOptions.put("Share", "ShareUntilDeath");
+    gameOptions.put("UnitCap", "500");
+    GameLaunchResponse gameLaunchMessage = GameLaunchMessageTestBuilder.create().defaultValues().team(1).expectedPlayers(4).mapPosition(3).gameOptions(gameOptions).get();
+    when(gameUpdater.update(any(), any(), any(), any())).thenReturn(completedFuture(null));
+    when(mapService.download(gameLaunchMessage.getMapName())).thenReturn(completedFuture(null));
+    when(fafService.startSearchMatchmaker()).thenReturn(completedFuture(gameLaunchMessage));
+    instance.startSearchMatchmaker();
+    verify(forgedAllianceService).startGame(
+        gameLaunchMessage.getUid(), null, List.of("/team", "1", "/players", "4", "/startspot", "3", "/gameoptions", "Share:ShareUntilDeath UnitCap:500"),
```

Comment:
Whats a good way to also allow `"UnitCap:500 Share:ShareUntilDeath"`?

Reply:
Does Java have an ordered map? You could use that instead of a HashMap.

Reply:
Seems like using LinkedHashMap in the kotlin file is the way to go.

Reply:
Does order actually matter? If it doesn't then I would not use a Linked Hash map.

There is a contains in any order matcher as well

Reply:
oh btw I think the game is expecting the args to look like `["/gameoption", "UnitCap:500", "Share:ShareUntilDeath"]` not `["/gameoption", "UnitCap:500 Share:ShareUntilDeath"]`

Reply:
What is the difference there they just get appended onto the command line arg with a space as a joiner. I would prefer the second as that follows the pattern of key and then value rather than having a multivalued key.

Reply:
On the verification you can also use an argument captor and do the verification after the verify call

Reply:
The difference is that one has 2 command line args while the other has 3. As it stands the game doesn’t try to split the args by space, just by colon. I mean maybe we could add that but imo just pass the args correctly.

Reply:
I think it would be better to keep it as a single key and single value to maintain the same pattern as all the other argument pairs.

Reply:
Currently the client splits all arguments at spaces: https://github.com/FAForever/downlords-faf-client/blob/221e3fb2309212167304c3977087bb652337b890/src/main/java/com/faforever/client/game/GameService.java#L612

Reply:
Maybe although I don't think that field is actually used but I guess it is okay